src/buffer/out/AttrRow.cpp

```diff
@@ -288,21 +288,40 @@ void ATTR_ROW::ReplaceAttrs(const TextAttribute& toBeReplacedAttr, const TextAtt
         // Check for that circumstance by seeing if we're inserting a single run of the
         // left side color right at the boundary and just adjust the counts in the existing
         // two elements in our internal list.
-        else if (_list.size() == 2 && newAttrs.at(0).GetLength() == 1)
+        else if (iStart == iEnd && iStart > 0)
         {
-            const auto left = _list.begin();
-            if (iStart == left->GetLength() && NewAttr == left->GetAttributes())
+            size_t lowerBound = 0;
+            size_t upperBound = 0;
+            for (auto i = 0; i < _list.size(); i++)

```

Comment:
Use of `auto `is good, elsewhere, but not in this case. To illustrate the point:

```
auto i = 0;  // deduces to int
auto i = 0u; // deduces to unsigned int 
```

The return type of `std::list::size()` is `std::size_t`. On x86 builds this is defined as `unsigned int`, but on x64 builds this is defined as `unsigned long long int`. So, when building x64, there is a difference of signed vs unsigned and 32-bit integer vs 64-bit integer. 

When using `auto` (deducing to int) this is comparing an `int` with and `unsigned long long int`. If the size of the list is > MAX_INT (unlikely to be the case in this particular scenario) this will become an infinite loop.

To prevent unwanted surprises, this is best rewritten as:

`for (size_t i = 0; i < _list.size(); i++)`

or

`for (std::size_t i = 0; i < _list.size(); i++)`

Reply:
Thanks! I came from a Java/C# background and didn't realize this until now.